src/digests/digest.lisp

```diff
@@ -177,17 +177,18 @@
              (let ((digest-size ,(if single-digest-p
                                      (second (first specs))
                                      `(etypecase state
-                                        ,@(reverse specs)))))
+                                        ,@(reverse specs))))
+                   (state-copy (copy-digest state)))
```

Comment:
Nearly every interesting digest doesn't preserve state through `produce-digest`:

```
CL-USER> (dolist (n (ironclad:list-all-digests))
	   (format t "~A gives same result from multiple PRODUCE-DIGEST calls: ~A~%"
		   n (check n)))
MD5 gives same result from multiple PRODUCE-DIGEST calls: NIL
RIPEMD-128 gives same result from multiple PRODUCE-DIGEST calls: NIL
SKEIN256/160 gives same result from multiple PRODUCE-DIGEST calls: NIL
SHA256 gives same result from multiple PRODUCE-DIGEST calls: NIL
SKEIN1024/512 gives same result from multiple PRODUCE-DIGEST calls: NIL
SKEIN512/256 gives same result from multiple PRODUCE-DIGEST calls: NIL
SKEIN256/224 gives same result from multiple PRODUCE-DIGEST calls: NIL
SHA224 gives same result from multiple PRODUCE-DIGEST calls: NIL
WHIRLPOOL gives same result from multiple PRODUCE-DIGEST calls: NIL
SKEIN1024/384 gives same result from multiple PRODUCE-DIGEST calls: NIL
SHA1 gives same result from multiple PRODUCE-DIGEST calls: NIL
MD2 gives same result from multiple PRODUCE-DIGEST calls: NIL
SKEIN512/384 gives same result from multiple PRODUCE-DIGEST calls: NIL
MD4 gives same result from multiple PRODUCE-DIGEST calls: NIL
SKEIN512/224 gives same result from multiple PRODUCE-DIGEST calls: NIL
SKEIN256 gives same result from multiple PRODUCE-DIGEST calls: NIL
SKEIN512/160 gives same result from multiple PRODUCE-DIGEST calls: NIL
ADLER32 gives same result from multiple PRODUCE-DIGEST calls: T
TIGER gives same result from multiple PRODUCE-DIGEST calls: NIL
SKEIN512 gives same result from multiple PRODUCE-DIGEST calls: NIL
SHA512 gives same result from multiple PRODUCE-DIGEST calls: NIL
SKEIN512/128 gives same result from multiple PRODUCE-DIGEST calls: NIL
SHA384 gives same result from multiple PRODUCE-DIGEST calls: NIL
SKEIN256/128 gives same result from multiple PRODUCE-DIGEST calls: NIL
CRC24 gives same result from multiple PRODUCE-DIGEST calls: T
CRC32 gives same result from multiple PRODUCE-DIGEST calls: T
SKEIN1024 gives same result from multiple PRODUCE-DIGEST calls: NIL
TREE-HASH gives same result from multiple PRODUCE-DIGEST calls: NIL
RIPEMD-160 gives same result from multiple PRODUCE-DIGEST calls: NIL
NIL
```

The documentation even explicitly states that `produce-digest` modifies the internal state of the given digest and suggests using `copy-digest` if you need to preserve state.